packages/react-components/src/shared/src/mergeProps.ts

```diff
@@ -27,33 +32,34 @@ export class CompositeKeyWeakMap {
             }
         }
 
-        node.value = value;
+        (node as Child<T>).value = value;
     }
 
-    get(keys) {
-        let node = this._root;
+    get(keys: any[]) {
+        let node: WeakMap<any, any> | Child<T> = this._root;
 
         for (let i = 0; i < keys.length; i += 1) {
-            node = (node.map || node).get(keys[i]);
+            const map: WeakMap<any, any> = (node as Child<T>).map || node as WeakMap<any, any>;
+            node = map.get(keys[i]);
 
             if (isUndefined(node)) {
                 return node;
             }
         }
 
-        return node.value;
+        return (node as Child<T>).value;
     }
 
-    has(keys) {
+    has(keys: any[]) {
         return !isUndefined(this.get(keys));
     }
 }
 
 ////////
 
-const cache = new CompositeKeyWeakMap();
+const cache = new CompositeKeyWeakMap<any>();
 
-function memoMerge(x, y, fct) {
+function memoMerge<TFirst, TSecond, TReturnValue>(x: TFirst, y: TSecond, fct: (y: TSecond, x: TFirst) => TReturnValue) {
```

Comment:
How come in "fct" x and y are reversed?

Reply:
TReturnValue => TReturn ?

Reply:
memoMerge was coded that way. It felt weird to me as well. 
if you look below, you can see 

    const mergeResult = fct(y, x);